docs/usage.rst

```diff
@@ -156,19 +156,24 @@ requests to a service's "search" endpoint. This method returns a
     ...     datetime=['2019-01-01T00:00:00Z', '2019-01-02T00:00:00Z'],
     ... )
 
-Instances of :class:`~pystac_client.ItemSearch` have 2 methods for iterating
-over results:
-
-* :meth:`ItemSearch.item_collections
-  <pystac_client.ItemSearch.item_collections>`: iterator over *pages* of results,
-  yielding an :class:`~pystac.ItemCollection` for each page of results.
-* :meth:`ItemSearch.items <pystac_client.ItemSearch.items>`: an iterator over
-  individual Item objects, yielding a :class:`pystac.Item` instance for all Items
-  that match the search criteria.
-* :meth:`ItemSearch.items_as_dicts <pystac_client.ItemSearch.items_as_dicts>`:
-  iterate over individual results, yielding a :class:`dict` instance representing each
-  item that matches the search criteria. This eliminates the overhead of creating
-  class:`pystac.Item` objects
+Instances of :class:`~pystac_client.ItemSearch` have a handful of methods for
+getting matching items into Python objects. The method to use depends on how
+many of the matches you want to consume (a single item at a time, a
+page at a time, or everything) and whether you want "raw" Python dictionaries
+representing the items, or proper ``pystac`` objects.
+
+* :meth:`ItemCollection.items()` returns an iterator over individual items,
+  each being a :class:`pystac.Item`.
+* :meth:`ItemCollection.items_as_dicts()` returns an iterator over individual
+  items, each being an item-like :class:`dict`.
+* :meth:`ItemCollection.pages()` returns an iterator over pages of the
+  matching items, each being a :class:`pystac.ItemCollection`
+* :meth:`ItemCollection.pages_as_dicts()` returns an iterator over pages of the
+  matching items, each being feature-collection-like :class:`dict`.
+* :meth:`ItemCollection.item_collection()` returns all of the matching items
+  as a single :class:`pystac.ItemCollection`
+* :meth:`ItemCollection.item_collection_as_dict()` returns  all of the matching
+  items as a single feature-collection-like :class:`dict`.
```

Comment:
Do you think a table like this would be useful, or only increase confusion?

|  | PySTAC object | Raw dictionary |
| -- | -- | -- |
| Iterator over one | `items` | `items_as_dicts` |
| Iterator over pages | `pages` | `pages_as_dicts` |
| Everything | `item_collection` | `item_collection_as_dict` |

Reply:
You caught me: I was going to add a table but I didn't want to look up the restructured text syntax :) I'll add it.

Reply:
Added a table. I needed to bump the maximum allowed line-length in the doc files to accommodate it since I didn't immediately see how to disable that check for tables.